src/libs/Navigation/linkTo.ts

```diff
@@ -1,72 +1,86 @@
 import NAVIGATORS from '@src/NAVIGATORS';
+import {Route} from '@src/ROUTES';
 import getStateFromPath from './getStateFromPath';
 import getTopmostReportId from './getTopmostReportId';
 import linkingConfig from './linkingConfig';
+import {NavigationRoot, RootStackParamList, StackNavigationAction} from './types';
+
+type ActionPayloadParams = {
+    screen?: string;
+    params?: unknown;
+    path?: string;
+};
+
+type ActionPayload = {
+    params?: ActionPayloadParams;
+};
 
 /**
  * Motivation for this function is described in NAVIGATION.md
  *
- * @param {Object} action action generated by getActionFromState
- * @param {Object} state The root state
- * @returns {Object} minimalAction minimal action is the action that we should dispatch
+ * @param action action generated by getActionFromState
+ * @param state The root state
+ * @returns minimalAction minimal action is the action that we should dispatch
  */
-function getMinimalAction(action, state) {
-    let currentAction = action;
-    let currentState = state;
-    let currentTargetKey = null;
+function getMinimalAction(action: NavigationAction, state: NavigationState): Writable<NavigationAction> {
+    let currentAction: NavigationAction = action;
+    let currentState: NavigationState | PartialState<NavigationState> | undefined = state;
+    let currentTargetKey: string | undefined;
 
-    while (currentState.routes[currentState.index].name === currentAction.payload.name) {
-        if (!currentState.routes[currentState.index].state) {
+    while (currentAction?.payload && 'name' in currentAction.payload && currentState?.routes[currentState?.index ?? -1].name === currentAction?.payload?.name) {
+        if (!currentState?.routes[currentState?.index ?? -1].state) {
             break;
         }
 
-        currentState = currentState.routes[currentState.index].state;
+        currentState = currentState?.routes[currentState?.index ?? -1].state;
+        currentTargetKey = currentState?.key;
 
-        currentTargetKey = currentState.key;
+        const payload = currentAction.payload as unknown as ActionPayload;
 
         // Creating new smaller action
         currentAction = {
             type: currentAction.type,
             payload: {
-                name: currentAction.payload.params.screen,
-                params: currentAction.payload.params.params,
-                path: currentAction.payload.params.path,
+                name: payload?.params?.screen,
+                params: payload?.params?.params,
+                path: payload?.params?.path,
             },
             target: currentTargetKey,
         };
     }
     return currentAction;
 }
 
-export default function linkTo(navigation, path, type, isActiveRoute) {
-    if (navigation === undefined) {
+export default function linkTo(navigation: NavigationContainerRef<RootStackParamList> | null, path: Route, type?: string, isActiveRoute?: boolean) {
+    if (navigation === undefined || navigation === null) {
```

Comment:
Is there any reason why you don't use just
```suggestion
    if (!navigation) {
```